### ISSUE TITLE
Cant create C# classes using XSD.exe

### ISSUE THREAD

**Issue by davidcorbin-atmosera** [doxygen/doxygen#10195]
**Describe the bug**
Attempting to use Microsoft Visual Studio Tool to generate classes from CSD files here in the repository.

**Expected behavior**
Tool runs without errors.

**Screenshots**
Microsoft (R) Xml Schemas/DataTypes support utility
[Microsoft (R) .NET Framework, Version 4.8.3928.0]
Copyright (C) Microsoft Corporation. All rights reserved.
Error: Error generating classes for schema 'C:\Usedoxygen --rs\david_corbin\source\repos\doxygen\templates\xml\compound'.
  - Group 'docTitleCmdGroup' from targetNamespace='' has invalid definition: Circular group reference.

**To Reproduce**
xsd /c compound.xsd

**Version**
Current clone of Repository  

**Stack trace**
N/A

**Additional context**

**Comment by albert-github** [comment]
- Current clone of Repository

This is a moving target so please specify the version used (though I can reproduce the problem with the current master, 1.9.8 (2777acd99c6bce2b2d0d1fa2c148c75bcf61a7fd) and an empty project).

The message:
> Group 'docTitleCmdGroup' from targetNamespace='' has invalid definition: Circular group reference.

It looks like the
```
  <xsd:group name="docCmdGroup">
    <xsd:choice>
      <xsd:group ref="docTitleCmdGroup"/>
```

so the group in the group or choice in choice gives some problems.

- what is the purpose of the the generation of the classes? Is it something similar to what is done for `doxmlparser`?

**Comment by davidcorbin-atmosera** [comment]
@albert-github - Thanks for the prompt response. First the easy part, I can confirm I am using 2777acd99c6bce2b2d0d1fa2c148c75bcf61a7fd 

As for "why", that is more complicated. I am looking to perform some analytics. Rather than attempting to parse source code, using the DoxyGen XML files seemed good (still does!), but then there was the choices in a .Net world of how to load them, and I was investingating XSD for strongly typed classes...

Not familiar with doxmlparse... will have to look that up.
I would love to discuss:  david(dot)corbin(at)atmosera(dot)com.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10196
(even though it is not really nice to copy the data)

**Comment by davidcorbin-atmosera** [comment]
Thanks. That gave me an idea (though I really wish MFST would update XSD to properly support Circular - and svcutil is not a great option either).

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
